docs/hiveutil.md

```diff
@@ -0,0 +1,47 @@
+# hiveutil
+
+The `hiveutil` CLI offers several commands to help manage clusters with Hive.
+
+To build the `hiveutil` binary, run `make hiveutil`.
+
+### Create Cluster
+
+The `create-cluster` command generates a `ClusterDeployment` and submits it to the Hive cluster using your current kubeconfig.
+
+To view what `create-cluster` generates, *without* submitting it to the API server, add `-o yaml` to the above command. If you need to make any changes not supported by `create-cluster` options, the output can be saved, edited, and then submitted with `oc apply`. This is also a useful way to generate sample yaml.
+
+By default, `create-cluster` assumes the latest Hive master CI build and the latest OpenShift stable release. `--release-image` can be specified to control which OpenShift release image to use.
+
+#### Create Cluster on AWS
+
+Credentials will be read from your AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY environment variables. Alternatively you can specify an AWS credentials file with --creds-file.
+
+```bash
+bin/hiveutil create-cluster --base-domain=mydomain.example.com --cloud=aws mycluster
+```
+
+#### Create Cluster on Azure
+
+Credentials will be read from `~/.azure/osServicePrincipal.json` typically created via the `az login` command.
+
+```bash
+bin/hiveutil create-cluster --base-domain=mydomain.example.com --cloud=azure --azure-base-domain-resource-group-name=myresourcegroup mycluster
+```
+
+#### Create Cluster on GCP
+
+Credentials will be read from `~/.gcp/osServiceAccount.json`, this can be created by:
```

Comment:
The creds could also come from the file pointed to be the GCP_SHARED_CREDENTIALS_FILE environment variable or the --creds-file argument.

Reply:
done